packages/plugin-core/src/survey.ts

```diff
@@ -0,0 +1,173 @@
+  }
+  onReject(_opts?: any): void {
+    return undefined;
+  }
+
+  async show() {
+    const results = await vscode.window.showQuickPick(this.choices, this.opts);
+    if (results) {
+      this.onAnswer(results);
+    } else {
+      this.onReject();
+    }
+
+    return results;
+  }
+}
+
+export class BackgroundSurvey extends DendronSurvey {
+  onAnswer(results: vscode.QuickPickItem[]) {
+    AnalyticsUtils.track(SurveyEvents.BackgroundAnswered, {
+      results: results.map((result) => result.label),
+    });
+  }
+
+  onReject() {
+    AnalyticsUtils.track(SurveyEvents.BackgroundRejected);
+  }
+
+  static create() {
+    const title = "What is your background?";
+    const choices = [
+      { label: "Software Developer" },
+      { label: "Technical Writer" },
+      { label: "Dev Ops" },
+      { label: "Manager" },
+      { label: "Student" },
+      { label: "Other" },
+    ];
+    return new BackgroundSurvey({ title, choices, canPickMany: true });
+  }
+}
+
+export class UseCaseSurvey extends DendronSurvey {
+  onAnswer(results: vscode.QuickPickItem[]) {
+    AnalyticsUtils.track(SurveyEvents.UseCaseAnswered, {
+      results: results.map((result) => result.label),
+    });
+  }
+
+  onReject() {
+    AnalyticsUtils.track(SurveyEvents.UseCaseRejected);
+  }
+
+  static create() {
+    const title = "Tell us what you want to use Dendron for";
+    const choices = [
+      { label: "Team knowledge base" },
+      { label: "Todos and Agenda" },
+      { label: "Meeting notes" },
+      { label: "Publishing" },
+      { label: "Research" },
+      { label: "Other" },
+    ];
+    return new UseCaseSurvey({ title, choices, canPickMany: true });
+  }
+}
+
+export class PriorToolsSurvey extends DendronSurvey {
+  onAnswer(results: vscode.QuickPickItem[]) {
+    AnalyticsUtils.track(SurveyEvents.PriorToolsAnswered, {
+      results: results.map((result) => result.label),
+    });
+  }
+
+  onReject() {
+    AnalyticsUtils.track(SurveyEvents.PriorToolsRejected);
+  }
+
+  static create() {
+    const title = "Are you coming from an existing tool(s)?";
+    const choices = [
+      { label: "No" },
+      { label: "Foam" },
+      { label: "Roam" },
+      { label: "Logseq" },
+      { label: "Notion" },
+      { label: "OneNote" },
+      { label: "Obsidian" },
+      { label: "Evernote" },
+      { label: "Google Keep" },
+    ];
+    return new PriorToolsSurvey({ title, choices, canPickMany: true });
+  }
+}
+
+export class SurveyUtils {
+  /**
+   * Flip a coin to randomly prompt initial survey.
+   * Asks three questions about background, use case, and prior tools used.
+   * @param forcePrompt skip flipping a coin and force prompting.
+   */
+  static async maybePromptInitialSurvey(forcePrompt?: boolean) {
+    const shouldPrompt = forcePrompt ? true : !!Math.floor(Math.random() * 2);
+    if (shouldPrompt) {
+      AnalyticsUtils.track(SurveyEvents.InitialSurveyPrompted);
+      vscode.window
+        .showInformationMessage(
```

Comment:
instead of using an information message, can we show a full on modal after the user clicks showTutorial?

Reply:
@kevinslin  - I think this is already set with `modal: true` prop below?